### ISSUE TITLE
Linking SPIRV-Tools with C code (shared/dynamic library request)

### ISSUE THREAD

**Issue by jozefkucia** [KhronosGroup/SPIRV-Tools#1240]
Using SPIRV-Tools in C code is a bit problematic.

Let's take as an example the following code:
```
#include <stdio.h>
#include "spirv-tools/libspirv.h"

int main(int argc, char **argv)
{
    spv_context context = spvContextCreate(SPV_ENV_VULKAN_1_0);
    spvContextDestroy(context);
    return 0;
}
```

The usual way of building it fails:
```
$ gcc main.c $(pkg-config --cflags --libs SPIRV-Tools)

/home/joseph/src/vulkan/install/lib64/libSPIRV-Tools.a(table.cpp.o): In function `spvContextCreate':
/home/joseph/src/vulkan/SPIRV-Tools/source/table.cpp:52: undefined reference to `operator new(unsigned long)'
/home/joseph/src/vulkan/install/lib64/libSPIRV-Tools.a(table.cpp.o): In function `spvContextDestroy':
/home/joseph/src/vulkan/SPIRV-Tools/source/table.cpp:55: undefined reference to `operator delete(void*)'

...

collect2: error: ld returned 1 exit status
```

I can fix this by linking with `libstdc++` or by using `g++`, but I would like to avoid this if possible.

Could SPIRV-Tools be also built as a shared library?

**Comment by dneto0** [comment]
Yes, we did something similar with Shaderc:
https://github.com/google/shaderc/blob/master/libshaderc/CMakeLists.txt#L17

**Comment by jozefkucia** [comment]
I can prepare a pull request, if that would be welcome.

**Comment by dneto0** [comment]
Fixed with #1263

**Comment by dneto0** [comment]
Related:
#1276 
#1275
